plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2932,17 +2932,21 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
                 .orElseThrow(() -> new IllegalStateException("Storage table missing in definition of materialized view " + materializedViewName));
 
         Table icebergTable = catalog.loadTable(session, storageTableName);
-        String dependsOnTables = Optional.ofNullable(icebergTable.currentSnapshot())
+        Optional<Snapshot> currentSnapshot = Optional.ofNullable(icebergTable.currentSnapshot());
+        String dependsOnTables = currentSnapshot
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
         if (dependsOnTables.isEmpty()) {
             // Information missing. While it's "unknown" whether storage is stale, we return "stale".
             // Normally dependsOnTables may be missing only when there was no refresh yet.
             return new MaterializedViewFreshness(STALE, Optional.empty());
         }
-        Instant refreshTime = Optional.ofNullable(icebergTable.currentSnapshot().summary().get(TRINO_QUERY_START_TIME))
+        Instant refreshTime = currentSnapshot.map(snapshot -> snapshot.summary().get(TRINO_QUERY_START_TIME))
                 .map(Instant::parse)
-                .orElseGet(() -> Instant.ofEpochMilli(icebergTable.currentSnapshot().timestampMillis()));
+                .or(() -> currentSnapshot.map(snapshot -> Instant.ofEpochMilli(snapshot.timestampMillis())))
+                // this will never fail because if dependsOnTables is not empty it means there is a table and
+                // thus currentSnapshot is not null so code in or branch will return a result
```

Comment:
i'm not sure i understand what is meant here

`// thus currentSnapshot is not null so code in or branch will return a result`

Reply:
Above this comment there is a check `if (dependsOnTables.isEmpty()) {` if `dependsOnTables` were empty we wouldn't get to the place in code with this comment. So if we get there it means `dependsOnTables` is not empty so it implicates that 'currentSnapshot` is not null so `orElseThrow` won't be called